perfschema/statement.go

```diff
@@ -0,0 +1,322 @@
+		state.info.key,           // EVENT_ID
+		nil,                      // END_EVENT_ID
+		state.info.name,          // EVENT_NAME
+		state.source,             // SOURCE
+		uint64(state.timerStart), // TIMER_START
+		uint64(state.timerEnd),   // TIMER_END
+		nil, // TIMER_WAIT
+		uint64(state.lockTime),             // LOCK_TIME
+		state.sqlText,                      // SQL_TEXT
+		nil,                                // DIGEST
+		nil,                                // DIGEST_TEXT
+		state.schemaName,                   // CURRENT_SCHEMA
+		nil,                                // OBJECT_TYPE
+		nil,                                // OBJECT_SCHEMA
+		nil,                                // OBJECT_NAME
+		nil,                                // OBJECT_INSTANCE_BEGIN
+		nil,                                // MYSQL_ERRNO,
+		nil,                                // RETURNED_SQLSTATE
+		nil,                                // MESSAGE_TEXT
+		uint64(state.errNum),               // ERRORS
+		uint64(state.warnNum),              // WARNINGS
+		state.rowsAffected,                 // ROWS_AFFECTED
+		state.rowsSent,                     // ROWS_SENT
+		state.rowsExamined,                 // ROWS_EXAMINED
+		uint64(state.createdTmpDiskTables), // CREATED_TMP_DISK_TABLES
+		uint64(state.createdTmpTables),     // CREATED_TMP_TABLES
+		uint64(state.selectFullJoin),       // SELECT_FULL_JOIN
+		uint64(state.selectFullRangeJoin),  // SELECT_FULL_RANGE_JOIN
+		uint64(state.selectRange),          // SELECT_RANGE
+		uint64(state.selectRangeCheck),     // SELECT_RANGE_CHECK
+		uint64(state.selectScan),           // SELECT_SCAN
+		uint64(state.sortMergePasses),      // SORT_MERGE_PASSES
+		uint64(state.sortRange),            // SORT_RANGE
+		uint64(state.sortRows),             // SORT_ROWS
+		uint64(state.sortScan),             // SORT_SCAN
+		uint64(state.noIndexUsed),          // NO_INDEX_USED
+		uint64(state.noGoodIndexUsed),      // NO_GOOD_INDEX_USED
+		nil, // NESTING_EVENT_ID
+		nil, // NESTING_EVENT_TYPE
+		nil, // NESTING_EVENT_LEVEL
+	}
+}
+
+func (ps *perfSchema) updateEventsStmtsCurrent(connID uint64, record []interface{}) error {
+	store := ps.stores[TableStmtsCurrent]
+
+	batch := pool.Get().(*leveldb.Batch)
+	defer func() {
+		batch.Reset()
+		pool.Put(batch)
+	}()
+
+	rawKey := []interface{}{connID}
+	key, err := codec.EncodeKey(nil, rawKey...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	val, err := codec.EncodeValue(nil, record...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	batch.Put(key, val)
+
+	err = store.Write(batch, nil)
+	return errors.Trace(err)
+}
+
+func (ps *perfSchema) appendEventsStmtsHistory(record []interface{}) error {
+	store := ps.stores[TableStmtsHistory]
+	lastLsn := atomic.AddUint64(ps.lsns[TableStmtsHistory], 1)
+
+	batch := pool.Get().(*leveldb.Batch)
+	defer func() {
+		batch.Reset()
+		pool.Put(batch)
+	}()
+
+	lsn := lastLsn - 1
+	rawKey := []interface{}{uint64(lsn)}
+	key, err := codec.EncodeKey(nil, rawKey...)
```

Comment:
`events_statements_current` table only keep one record for each connection (most recent), `events_statements_history` table just keep all records (maximum 1024 so far), so here the key is just LSN, that is, we simply increment this number for each append operation, this is the same as other setup_xxx tables.